modules/aws/lls/lls-startup.sh.tmpl

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# Copyright (c) 2020 Teradici Corporation
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+
+INST_LOG_FILE="/var/log/teradici/lls/install.log"
+LLS_YUM_PKG="pcoip-license-server"
+
+log() {
+    local message="$1"
+    echo "[$(date)] $${message}" | tee -a "$INST_LOG_FILE"
+}
+
+get_credentials() {
+    if [[ -z "${customer_master_key_id}" ]]
+    then
+        log "Not using encryption"
+
+        LLS_ADMIN_PASSWORD=${lls_admin_password}
+        LLS_ACTIVATION_CODE=${lls_activation_code}
+
+    else
+        log "Using encryption key ${customer_master_key_id}"
+
+        LLS_ADMIN_PASSWORD=$(aws kms decrypt --region ${aws_region} --ciphertext-blob fileb://<(echo "${lls_admin_password}" | base64 -d) --output text --query Plaintext | base64 -d)
+        LL_ACTIVATION_CODE=$(aws kms decrypt --region ${aws_region} --ciphertext-blob fileb://<(echo "${lls_activation_code}" | base64 -d) --output text --query Plaintext | base64 -d)
+    fi
+
+    # Exit if any of the required variables are missing
+    if [[ -z "$LLS_ADMIN_PASSWORD" || -z "$LLS_ACTIVATION_CODE" ]]
+    then
+        log "Missing required parameters:"
+        exit 1
+    fi
+}
```

Comment:
Should I apply the same style of clean up changes from AWS provisioning to this startup script as well?

Reply:
not sure what you mean by same style of clean up.